### ISSUE TITLE
Hardsuit helmets survive the destruction of their suit with NODROP flag.

### ISSUE THREAD

**Issue by Ricotez** [tgstation/-tg-station#10226]
Explosions can now destroy clothing, and if a hardsuit is destroyed, its helmet can survive the blast. But the helmet still has the NODROP flag set, so the player can no longer remove the helmet if that happens.

**Comment by WJohn** [comment]
@Ikarrus

**Comment by Ikarrus** [comment]
I'm not sure how this is happening. 

The code forces the hardsuit to drop before destroying it, and unequipping the hardsuit should also unequip the helmet.

**Comment by Ricotez** [comment]
Invisity is the one who reported the bug to me, I asked him if he could post his story in this issue but he hasn't yet.

**Comment by phil235** [comment]
Deleting the hardsuit manually also deletes the helmet properly, so if there's an issue if it's not with /Destroy() per se.

**Comment by Ikarrus** [comment]
>Deleting the hardsuit manually also deletes the helmet properly, so if there's an issue if it's not with /Destroy() per se.

I just tried deleting the hardsuit and the helmet remained, so something is wrong here.

**Comment by xxalpha** [comment]
>drop issues

Ahem.

**Comment by xxalpha** [comment]
I don't think this is caused by that PR of mine.

**Comment by freerealestate** [comment]
It'd be nice if the helmet survived, if you could wear it independently of
the suit (i.e. the NODROP flag is removed).

**Comment by Ikarrus** [comment]
Yeah, I can't reproduce this reliably.

**Comment by Invisty** [comment]
Sorry for not posting here sooner.

What happened was that I set off a bomb in my vicinity while wearing a hardsuit, and the hardsuit was ripped off my body and destroyed. The helmet fell to the ground, and I picked it up and put it on.

I was then unable to remove it, since it can't be removed without clicking on the (non existent) hardsuit helmet toggle (which is associated with the hardsuit, not the helmet).

That's how I understand it.
